src/Meshes/Meshes.jl

```diff
@@ -164,6 +170,34 @@ function Mesh(; nodes::Vector{N}=Vector{AbstractNode}(),
     Mesh(nodes, elements, faces, node_sets, element_sets, face_sets, extra)
 end
 
+"Return the mesh node coordinates matrix. Each row is a node, each column a coordinate."
+function node_coordinates_matrix(mesh::Mesh{dim,T}) where {dim,T}
+    nodes_coords_matrix = Matrix{Float64}(undef, (num_nodes(mesh), dim))
+    for (i, n) in enumerate(nodes(mesh))
+        nodes_coords_matrix[i, :] = coordinates(n)
+    end
+    nodes_coords_matrix
+end
+
+"Return the mesh connectivity."
+function connectivity(mesh::Mesh{dim,T}) where {dim,T}
+
+    # Check if a already contains the connectivity
+    hasproperty(mesh.extra, :connectivity) && return mesh.extra.connectivity
```

Comment:
should we not cache the matrix at the end of this method as well?

Reply:
Do you mean the node_matrix ? I don't see why

Reply:
L186 checks if the connectivity matrix is already in `extra`, but where do you add it in the first place? im suggesting to add it before returning.

Reply:
Oh yes, got it. The first time in the case or gmsh meshes is added when the .gmsh is red. But for non gmsh meshes I agree that we can cache it. It is worth noting that this method has no performance effects is called for post-process.

Reply:
ok, we can leave it as is, as you prefer.